### PR TITLE
Break topic 2 into one file per subtopic. 

### PR DESCRIPTION
**Pull request by janetriley**
Issue #16 : fix error messages from running check.py against 02-func.md .  The primary issue was several topics in one file. 

Broke 02-func.md into separate files
Broke up learning objectives among new topics
Add rendered html 
Merged 'creating functions' and 'composition' into one topic, per issue discussion
Moved the boiling temp error into debugging  topic
Renumbered old topic files and  inserted new topic files;  updated index to match.

### CODE REVIEW

File: reference.html [hunk 1]
```diff
@@ -116,157 +116,157 @@ <h2 id="errors-and-exceptions"><a href="07-errors.html">Errors and Exceptions</a
 <h2 id="glossary">Glossary</h2>
```
Comment (janetriley) [inline reference.html]: Not sure why `make preview` rendered this differently than the version in the repository, but I expect it's correct.

**Comment by gvwilson** [comment]
Does this repo now get a clean bill of health from the checker?

**Comment by kynan** [comment]
@gvwilson No, though this pr fixes the issues in chapter 2 (but it no longer merges cleanly).

**Comment by janetriley** [comment]
I'll update and resubmit tonight or tomorrow.
Janet

**Comment by janetriley** [comment]
This has diverged, and I can't easily follow the merge conflicts. I'll re-do it off a current copy of the repo.

**Pull request closed without merging**